docs/rfcs/0000-persistent-logging-component.md

```diff
@@ -0,0 +1,111 @@
+
+From there, a user can share the logging via the UI or notification.
+
+### User Interface
+
+The UI would list all the recordings in the database that you can select to see the full logs for that record.
+
+In the UI for a selected recording, the rows of data collected with options to filter by the various attributes and their combinations (a description of these types can be see in the Table Schema section):
+
+ - recording ID
+ - tag
+ - severity
+
+The logs can then be shared with the Android system share service to be attached to an email, Github issue, clipboard, etc. This possibly not ideal, see the Drawbacks section for more details.
+
+Similar to `lib-crash` most of the basic UI will be written into an abtract activity/fragment.
+
+## Reference-level explanation
+
+### Table Schema
+
+The schema below describes the primary log table that will be used to store logs:
+
+| Column       | Description                                                  | Type    |
+| ------------ | ------------------------------------------------------------ | ------- |
+| id           | Primary key for each log line.                               | Long    |
+| timestamp    | The time when the log event had occurred.                    | Long    |
+| recording_id | A random UUID generated at the start of a log recording.     | String  |
+| severity     | The priority of the log statement. See Android's [Log](https://developer.android.com/reference/android/util/Log#ERROR) constants. | Char    |
+| tag          | Used to identify the source of a log message.  It usually identifies the class or activity where the log call occurs. | String? |
+| log          | The message you would like logged.                           | String? |
+| throwable    | Optional. An exception to log.                               | String? |
+
+A secondary table is for storing metadata for each recording:
+
+| Column          | Description                            | Type |
+| --------------- | -------------------------------------- | ---- |
+| recording_id    | Primary key; foreign key for log table | Long |
+| start_timestamp | When the recording was started.        | Long |
+| end_timestamp   | When the recording was ended.          | Long |
+
+
+
+## Drawbacks
+
+- Sharing logs through an intent is fallible and can throw exceptions if the contents are more than the Android OS can handle.
+- User-initiated recording would mean we can lose logs that are not easily reproducible. We can always re-consider changing this in the future.
+
+## Future Work
+
+- We can request an API to stream GeckoView logs to our `LogSink` .
```

Comment:
Ah, you mention this here. I expect this to not be feasible. Some time ago I asked about us having more control over logging and it seemed not easy to change since a bunch of things in Gecko may log in various parts of the code base.

Reply:
That's unfortunate. Maybe logs from the GeckoView API layer might benefit GV engineers?

Reply:
Hm sadness :(. I also think this would be the best solution for GV logs to go through our LogSink. Either way I think we should discuss this again with the GV team.

Reply:
Maybe not easy, but this is something we can probably address incrementally within GV? Agree with @csadilek that we should discuss this with GV again.